src/dbus.rs

```diff
@@ -149,12 +150,63 @@ pub struct Dbus {
 
 impl Dbus {
     pub fn new(eng: &Rc<AsyncEngine>, ring: &Rc<IoUring>, run_toplevel: &Rc<RunToplevel>) -> Self {
-        let user_path = match xrd() {
-            Some(path) => Some(format!("{}/bus", path)),
-            _ => {
-                log::warn!("{} is not set", XRD);
-                None
+        // https://dbus.freedesktop.org/doc/dbus-specification.html#addresses
+        fn unescape_value(escaped: &str) -> Option<String> {
+            let mut unescaped = Vec::new();
+            let mut bytes = escaped.bytes();
+            while let Some(c) = bytes.next() {
+                match c {
+                    b'-'
+                    | b'0'..=b'9'
+                    | b'A'..=b'Z'
+                    | b'a'..=b'z'
+                    | b'_'
+                    | b'/'
+                    | b'.'
+                    | b'\\'
+                    | b'*' => {
+                        unescaped.push(c);
+                    }
+                    b'%' => {
+                        let hi = (bytes.next()? as char).to_digit(16)?;
+                        let lo = (bytes.next()? as char).to_digit(16)?;
+                        unescaped.push((hi << 4 | lo) as u8);
+                    }
+                    _ => return None,
+                }
+            }
+            String::from_utf8(unescaped).ok()
+        }
+        let user_path = 'path: {
+            let Some(addr) = std::env::var(DBUS_SESSION_BUS_ADDRESS).ok() else {
+                log::warn!("{DBUS_SESSION_BUS_ADDRESS} is not set");
+                break 'path None;
```

Comment:
Please add a fallback to the original logic here.

Reply:
I can do that. It doesn't seem necessary, though - zbus doesn't do it, so maybe it's only necessary for broken setups.

Reply:
Sorry, I was wrong in my last comment. Will definitely add the fallback